### PR TITLE
Purge some additional jets

### PR DESCRIPTION
**Pull request by ojura**
Depends on #998 and #999.

### CODE REVIEW

**Comment by ojura** [comment]
@pifon2a Here are some additional uses of jets I think we can remove.

**Comment by ojura** [comment]
Ahh, crap. Seems like the changes in this PR need Eigen 3.3.

**Comment by ojura** [comment]
@pifon2a I got this branch to pass CI :) (It includes #998, with the change in acceleration_cost_function_3d.h reverted.)

File: cartographer/mapping/internal/3d/scan_matching/rotation_delta_cost_functor_3d.h [hunk 1]
```diff
@@ -42,11 +43,8 @@ class RotationDeltaCostFunctor3D {
   template <typename T>
   bool operator()(const T* const rotation_quaternion, T* residual) const {
     T delta[4];
```
Comment (pifon2a) [inline cartographer/mapping/internal/3d/scan_matching/rotation_delta_cost_functor_3d.h]: std::array<T, 4> delta;

File: cartographer/common/math.h [hunk 2]
```diff
@@ -74,6 +74,14 @@ T atan2(const Eigen::Matrix<T, 2, 1>& vector) {
   return ceres::atan2(vector.y(), vector.x());
 }
 
+template <typename T>
+inline void QuaternionProduct(const double z[4], const T w[4], T zw[4]) {
```
[the remark this answers is not in the document]
Comment (ojura) [reply cartographer/common/math.h]: Ok. I copy pasted this from Ceres, though. Are they using a different coding style?

**Comment by ojura** [comment]
@pifon2a updated

**Pull request merged**